### PR TITLE
[After #102] [Example] Add edc layout to the scaffolding @open sesame 05/18 16:27

### PR DESCRIPTION
**Pull request by zhoonit**
Add main.edc to separate UI design/interactions

**Changes proposed in this PR:**
- Add main.edc
- Add view.[ch]
- Add data.h
- Change main initiation to use view function
- Move data definitions to data.h

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: Applications/Tizen_native/CustomShortcut/src/view.c [hunk 1]
```diff
@@ -27,16 +33,93 @@ int view_create(widget_context_h context, int w, int h) {
...
+  int status = WIDGET_ERROR_NONE;
+
+  widget_app_context_get_tag(context, (void**) &wid);
+  wid->layout = _create_layout(wid->naviframe, wid->edj_path, "main", NULL, NULL);
+
+  if (wid->layout == NULL) {
+		dlog_print(DLOG_ERROR, LOG_TAG, "failed to create a layout of no alarm.");
+		evas_object_del(wid->win);
+		return WIDGET_ERROR_FAULT;
+	}
```
Comment (dongju-chae) [inline Applications/Tizen_native/CustomShortcut/src/view.c:88]: need indent fix

File: Applications/Tizen_native/CustomShortcut/src/view.c [hunk 2]
```diff
@@ -27,16 +33,93 @@ int view_create(widget_context_h context, int w, int h) {
...
+
+  if (parent == NULL){
+    dlog_print(DLOG_ERROR, LOG_TAG, "parent cannot be NULL");
+    return NULL;
+  }
+
+  layout = elm_layout_add(parent);
+  elm_layout_file_set(layout, edj_path, group_name);
+
+  if(layout == NULL){
```
Comment (dongju-chae) [inline Applications/Tizen_native/CustomShortcut/src/view.c:113]: need a space

File: Applications/Tizen_native/CustomShortcut/src/view.c (hunk 1 above)
Comment (zhoonit) [reply Applications/Tizen_native/CustomShortcut/src/view.c]: Oops, I should run clang format before push. Thanks :)

Comment (zhoonit) [reply Applications/Tizen_native/CustomShortcut/src/view.c]: Oops, I should run clang format before push. Thanks :)

File: Applications/Tizen_native/CustomShortcut/src/view.c (hunk 2 above)
Comment (zhoonit) [reply Applications/Tizen_native/CustomShortcut/src/view.c]: Fixed. Thanks!

File: Applications/Tizen_native/CustomShortcut/src/main.c [hunk 3]
```diff
@@ -29,18 +29,20 @@ static int widget_instance_create(widget_context_h context, bundle *content,
...
-  evas_object_resize(wid->label, w, h / 3);
-  evas_object_move(wid->label, w / 4, h / 3);
-  evas_object_show(wid->label);
-  elm_object_text_set(wid->label, "Hello widget!");
+  char *res_path = app_get_resource_path();
+  if (res_path) {
+    snprintf(wid->edj_path, sizeof(wid->edj_path), "%s%s", res_path, EDJ_PATH);
+    free(res_path);
+    dlog_print(DLOG_INFO, LOG_TAG, "file path: %s", wid->edj_path);
+  }
```
Comment (dongju-chae) [inline Applications/Tizen_native/CustomShortcut/src/main.c:43]: Q. is it okay not to return an error value when `res_path == nullptr`?

Comment (zhoonit) [reply Applications/Tizen_native/CustomShortcut/src/main.c]: Oops. Thanks for that :) It is now handled.

**Comment by jijoongmoon** [comment]
Need to resolve the conflicts.

**Comment by zhoonit** [comment]
> Need to resolve the conflicts.

Thanks :) Conflict resolved!

**Pull request merged**